pkg/console/operator/status.go

```diff
@@ -0,0 +1,299 @@
+//    example: during the initial deployment, the operator needs to create a number
+//      of resources.  the console (operand) is likely in flux.
+//    example: the assigned route .spec.host changes for the console.  the oauthclient
+//      must be updated, as must the configmap, etc.  numerous resources are in flux,
+//      the operator reports progressing until the resources stabilize
+// Failing = the operator (not the operand) is failing
+//    example: The console operator is unable to update the console config with
+//      a new logoutRedirect URL. The operator is failing to do its job, however the
+//      console (operand) may or may not be functional (see Available above).
+//
+//
+// Status Condition Reason & Message
+// Reason:  OperatorSyncLoopError
+// Message: "The operator sync loop was not completed successfully"
+//
+//
+
+const (
+	reasonUnmanaged           = "ManagementStateUnmanaged"
+	reasonRemoved             = "ManagementStateRemoved"
+	reasonSyncLoopProgressing = "SyncLoopProgressing"
+	reasonNoPodsAvailable     = "NoPodsAvailable"
+	reasonSyncError           = "SyncError"
+)
+
+// Lets transition to using this, and get the repetition out of all of the above.
+func (c *consoleOperator) SyncStatus(operatorConfig *operatorsv1.Console) (*operatorsv1.Console, error) {
+	logConditions(operatorConfig.Status.Conditions)
+	updatedConfig, err := c.operatorConfigClient.UpdateStatus(operatorConfig)
+	if err != nil {
+		return nil, fmt.Errorf("status update error: %v \n", err)
+	}
+	return updatedConfig, nil
+}
+
+// Outputs the condition as a log message based on the detail of the condition in the form of:
+//   Status.Condition.<Condition>: <Bool>
+//   Status.Condition.<Condition>: <Bool> (<Reason>)
+//   Status.Condition.<Condition>: <Bool> (<Reason>) <Message>
+//   Status.Condition.<Condition>: <Bool> <Message>
+func logConditions(conditions []operatorsv1.OperatorCondition) {
+	logrus.Println("Operator.Status.Conditions")
+	for _, condition := range conditions {
+		buf := bytes.Buffer{}
+		buf.WriteString(fmt.Sprintf("Status.Condition.%s: %s", condition.Type, condition.Status))
+		hasMessage := condition.Message != ""
+		hasReason := condition.Reason != ""
+		if hasMessage && hasReason {
+			buf.WriteString(" |")
+			if hasReason {
+				buf.WriteString(fmt.Sprintf(" (%s)", condition.Reason))
+			}
+			if hasMessage {
+				buf.WriteString(fmt.Sprintf(" %s", condition.Message))
+			}
+		}
+		logrus.Println(buf.String())
+	}
+}
+
+// setStatusCondition
+// A generic helper for setting a status condition.
+// To use when another more specific status function is not sufficient.
+// examples:
+//  setStatusCondition(operatorConfig, Failing, True, "SyncLoopError", "Sync loop failed to complete successfully")
+func (c *consoleOperator) SetStatusCondition(operatorConfig *operatorsv1.Console, conditionType string, conditionStatus operatorsv1.ConditionStatus, conditionReason string, conditionMessage string) *operatorsv1.Console {
```

Comment:
I don't see how this is better than calling `v1helpers.SetOperatorCondition` directly. It actually seems worse to me since it would be easy to mix up the order of the arguments.

Reply:
I also don't see where this is used.

Reply:
Its not used at this point, its been factored out.
I'll remove it.

Reply:
It was a step to initially condense the inline noise:

```golang
if aBadThing {
       logrus.Errorf("Bad things are happening")
       // logic....
	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
		Type:               BadType,
		Status:             BadStatus,
		Reason:             "ABadThing",
		Message:            "a bad thing",
		LastTransitionTime: metav1.Now(),
	})
       // oh my if more than one....
       // do other logic
}
// to a one-liner
setStatusCondition(operatorConfig, Failing, True, "SyncLoopError", "Sync loop failed to complete successfully")
```
That said, I agree, it was still too many things to take care of.  Thats when I moved to 

```golang
// one condition or 3, still one line... not 7,14,21, etc.
co.ConditionABadThing(operatorConfig)
```

Reply:
The right way to do this is probably for:

```golang
// this to return enough info:
structuredThing, err := sync_v400()
// that we could instead handle it all in one place in operator.go
handleConditions(structuredThing, err)
```
Tech debt...